Dockerfile

```diff
@@ -21,11 +25,14 @@ ADD LICENSE LICENSE
 COPY contrib/devtools/Makefile contrib/devtools/Makefile
 COPY Makefile .
 
+COPY go.mod .
+COPY go.sum .
 
-RUN make build
+RUN go mod download
 
-FROM ubuntu:22.04
+RUN BUILD_TAGS=muslc LINK_STATICALLY=true make build
```

Comment:
Some more context. Statically linking is not necessary, we could also do dynamic linking here and just grab the `.so` libs from wasmvm's `internal/api` folder (this is how I did it in the now closed PR). That is generally a bit cleaner since we don't need to reference the specific version of `libwasmvm` to download (i.e, we download `1.2.1` here to match what we have in our `go.mod`).

The downside is that, since we use multi-stage builds, any shared libraries we reference in the first stage of the build (on the `builder` image) must be copied over. We could also have a similar situations with other shared libs in the future. (I.e we'd need an additional `COPY --from=builder shared/lib/path` to reference them).

I don't particularly think it makes sense to use shared libraries in a docker image that won't have many components actually sharing the libraries but I don't have that strong of an opinion on it.